code/game/objects/structures/mirror.dm

```diff
@@ -233,20 +248,48 @@
 						H.dna.update_ui_block(DNA_FACIAL_HAIR_COLOR_BLOCK)
 				H.update_hair()
 
-		if(BODY_ZONE_PRECISE_EYES)
+		if("eyes")
 			var/new_eye_color = tgui_color_picker(H, "Choose your eye color", "Eye Color","#"+H.eye_color)
 			if(!user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
 				return
 			if(new_eye_color)
 				H.eye_color = sanitize_hexcolor(new_eye_color)
 				H.dna.update_ui_block(DNA_EYE_COLOR_BLOCK)
 				H.update_body()
+
+
+		// become a character from your prefs that you choose
+		if("check your character list")
+			var/list/character_list = user.client.prefs.get_sanitized_character_list()
+			if(!length(character_list))
+				return
+			var/character_choice = input(user, "Select a character to copy its appearance to you", "Magical Grooming: Character list") as null|anything in character_list
+			if(!character_choice)
+				return
+			var/slot_value = character_list[character_choice]
+			if(!slot_value)
+				return
+			var/old_slot_value = user.client.prefs.default_slot
+			if(!user.client.prefs.load_character(slot_value))
+				return
+			if(slot_value != old_slot_value) // it'll be annoying if your choice is unnoticingly changed
+				addtimer(CALLBACK(user.client.prefs, TYPE_PROC_REF(/datum/preferences, load_character), old_slot_value), 5 SECONDS, TIMER_UNIQUE|TIMER_OVERRIDE)
```

Comment:
This feels extremely abritrary

Reply:
It's a kind of hardcoding to solve UX issue. Your selected character is only loaded to the game, so the code should force-select your desired character through this, but doing this means your selected character is also changed unnoticingly.
Imagine you chose 2nd character as a wizard, and you join a next round as a crew but you accidentally become 2nd charaacter - especially 2nd character is a dedicated preset for a wizard character with awkward name and appearance.
The reason why it overrides the timer is that to make the preference code less costing by delaying the revert.

Reply:
Surely theres a way of loading a character from the DB without switching to it

Reply:
I am not competent to find a solution. I will just label my PR stale.